.github/workflows/int-and-connected-test-run.yml

```diff
@@ -74,14 +76,43 @@ jobs:
           base64 --decode <<< ${{ secrets.SA_B64_CREDENTIALS }} > ${GOOGLE_APPLICATION_CREDENTIALS}
           base64 --decode <<< ${{ secrets.B64_RBS_APPLICATION_CREDENTIALS }} > ${RBS_CLIENT_CREDENTIAL_FILE_PATH}
           # assemble code and run connected tests
-          ./gradlew assemble
-          ./gradlew testConnected --scan
+          ./gradlew --scan assemble testConnected
+  test_smoke:
+    name: Smoke tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'gradle'
+      - name: Run Test Runner smoke tests
+        env:
+          # required for smoke tests
+          GOOGLE_APPLICATION_CREDENTIALS: jade-dev-account.json
+          TEST_RUNNER_SA_KEY_DIRECTORY_PATH: '..'
+          # output plain logs instead of json
+          TDR_LOG_APPENDER: 'Console-Standard'
+        run: |
+          # extract service account credentials
+          base64 --decode <<< ${{ secrets.SA_B64_CREDENTIALS }} > ${GOOGLE_APPLICATION_CREDENTIALS}
+          # build the datarepo-client subproject
+          ENABLE_SUBPROJECT_TASKS=1 ./gradlew :datarepo-client:clean :datarepo-client:assemble
+          # find the datarepo-client jar from the clienttests subproject
+          # the odd capitalization of the environment variable is required
+          cd datarepo-clienttests
+          export ORG_GRADLE_PROJECT_datarepoclientjar=$(find .. -type f -name "datarepo-client*.jar")
+          # assemble code and run smoke tests
+          sed -i 's/"skipKubernetes": false/"skipKubernetes": true/' src/main/resources/servers/integration-6.json
+          ./gradlew spotlessCheck spotbugsMain runTest --args="suites/PRSmokeTests.json /tmp/TestRunnerResults"
```

Comment:
We talked about removing these smoke tests since we don't think they have a lot of value when running the integration tests locally. They also depend on Test Runner, which is not really maintained. If we really wanted to keep smoke tests, it would be better if we had a localhost profile that didn't depend on someone's personal environment.

Reply:
Agreed, I'm in favor of removing them from PR checks in this PR.

Reply:
I removed them in the latest version.

Reply:
Interesting, I would've advocated to keep the smoke test by using a localhost server config, so it works the same as the integration tests.

As I understand it, the idea was the smoke test would fail faster than the integration tests so it was useful to run it first. In a recent run, I see that the smoke tests took 12m and the integration tests took 1:30m.

> They also depend on Test Runner, which is not really maintained

I don't see that as being relevant; the test runner here is not the terra test runner. It's part of the TDR code base, so it's as maintained as anything else is in this repo.

Reply:
> As I understand it, the idea was the smoke test would fail faster than the integration tests so it was useful to run it first. In a recent run, I see that the smoke tests took 12m and the integration tests took 1:30m.

Where I've landed on this is that from my vantage point, PR smoke tests have very rarely caught an issue that would have otherwise been exposed only at the end of integration tests.  Unit tests for the most part serve this role.  The vast majority of the time, PR smoke test failures have been flaky connection issues that succeeded on a rerun (but annoyed me in the process).

If we were to keep them on PRs in some form, I would want them to run concurrently with integration tests, rather than serially.  I understand why they were serially chained, but the vast majority of the time I don't think that the extra 12-20 minutes in overall test runtime is a good trade-off.

> the test runner here is not the terra test runner.

🤯 wow, I've been walking around assuming that we were using the Terra Test Runner library, and you're right, we're not.  I actually think this might be worse, because I *know* that TDR's Test Runner has not been maintained and is not a standard followed by other services for testing. https://github.com/DataBiosphere/jade-data-repo/tree/b9e99202f1f0c9ba8547df3b8dfb4801fcdd75ba/datarepo-clienttests